solutions/usaco-596.mdx

```diff
@@ -93,7 +93,62 @@ void ff(int i, int j)
 	}
 }
 ```
+## Alternate Faster Solution
 
+It is possible to make the following 2 observations:
+
+1. For each contiguous closed region, there must be 1 gate. In a minimum state, each region has exactly 1 gate.
+2. Each time FJ comes across a node he has visited along an edge he has not crossed in either direction, he creates exactly 1 new region. In other words, if FJ walks along an edge from point a to b, and he did not walk from a to b in the past, and he did not walk from b to a in the past, and b is a point he had already visited, then he has created exactly 1 new region.
```

Comment:
this is a run on sentence. Maybe consider breaking it as a bulleted list and saying these are the conditions that must be satisfied.

Reply:
Done